82. Remove Duplicates from Sorted List II.md

olsen-blue opened the pull request:
・問題
https://leetcode.com/problems/remove-duplicates-from-sorted-list-ii/description/

・次に解く問題
https://leetcode.com/problems/add-two-numbers/description/

olsen-blue committed:
Create 82. Remove Duplicates from Sorted List II.md

olsen-blue committed:
Update 82. Remove Duplicates from Sorted List II.md

olsen-blue committed:
Update 82. Remove Duplicates from Sorted List II.md

olsen-blue committed:
Update 82. Remove Duplicates from Sorted List II.md

olsen-blue committed:
Update 82. Remove Duplicates from Sorted List II.md

oda commented on 82. Remove Duplicates from Sorted List II.md:
```diff
@@ -0,0 +1,112 @@
+class Solution:
+    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
+        dummy = ListNode(0, head)
+        flag = dummy
```
flag はブール値を期待しますね。これ、意味合いとしては、dummy から繋がっていて出力されることが確定したものの最後ですよね。

oda replied:
そういう意味で、flag.next には、これからチェックされる node が常にいますが、私の気持ちとしては、flag.next = None のままにしておいて、flag.next = node; flag = flag.next; flag.next = None とするほうが私の気持ちには近いです。(ただ、node = node.next の更新を先にしないといけないので少し説明が面倒です。)

olsen-blue replied:
コメントいただきましてありがとうございます。
仰る通り「flag」だと、フラグが「立つか」「立たないか」なので、boolのイメージが近いですね。

他のコードでは「tail」という変数名を見た記憶があったので、まず「tail_fixed_node」とかでどうかなと思いましたが、仰る通り、「tail」と言っているのに、tail_fixed_node.next に、これからチェックされる node が常にいるのが、なんだか直感的ではないなと感じました。
最終的には「last_fixed_node」どうかなと思い、そのように修正してみました。

olsen-blue committed:
flagの変数名修正

odaさんのコメント
https://discord.com/channels/1084280443945353267/1322513618217996338/1324484932164128809

olsen-blue committed:
空行除去、dummyに値を与えない、という修正

syoshida-nb commented on 82. Remove Duplicates from Sorted List II.md:
```diff
@@ -0,0 +1,160 @@
+        last_fixed_node = dummy
+        node = head
+
+        while node is not None and node.next is not None:
```
L130で、nos.next is not Noneを確認するので、こちらのIF文では `node.next is not None` が不要だと思いました。
